desicochrane/basic-mailbox-api
# Add basic command line tool to import messages

# Why does this PR exist?

This adds a command line tool to import messages from a json file.

### How should this be manually tested?

1. get a copy of `messages_sample.json` or any json file with the following structure:

    ```js
    {
        "messages": [
            {
                "uid": "21",
                "sender": "Ernest Hemingway",
                "subject": "animals",
                "message": "This is a tale about nihilism. The story is about a combative nuclear...", 
                "time_sent": 1459239867
            },
            // etc...
        ]
    }    
    ```
1. run `php artisan mail-api:import <path-to-your-json-file>`
1. check the db and see your data has been imported

### Discussion

This was done as a kind of once-off throw-away command specific to the sample file given.

There are many `//TODO`s in the file for improvements that could be made if this import tool needed to be more defensible and robust or work with very large files.

No tests have been added for this command due it being once-off throwaway in nature.
